src/vtcomposite.cpp

```diff
@@ -370,7 +385,46 @@ Napi::Value composite(Napi::CallbackInfo const& info)
             return utils::CallbackError("'y' value must not be less than zero", info);
         }
 
-        baton_data->tiles.push_back(std::make_unique<TileObject>(z, x, y, buffer));
+        // layers array value
+        // does the layers key exist?
+        std::vector<std::string> layers;
+        if (tile_obj.Has(Napi::String::New(info.Env(), "layers")))
+        {
+            Napi::Value layers_val = tile_obj.Get(Napi::String::New(info.Env(), "layers"));
+
+            // is the layers property an array?
+            if (!layers_val.IsArray())
+            {
+                return utils::CallbackError("'layers' value in the 'tiles' array must be an array", info);
+            }
+
+            Napi::Array layers_array = layers_val.As<Napi::Array>();
+            std::uint32_t num_layers = layers_array.Length();
+            // does the layers array have length > 0?
+            if (num_layers <= 0)
```

Comment:
Since `num_layers` is unsigned, shouldn't it be `== 0`?

Reply:
Yeah we can probably just use `==` - I was trying to capture any unexpected situations here but the length of an Napi cannot be negative, presumably.

Reply:
resolved in https://github.com/mapbox/vtcomposite/pull/113/commits/e4720334e9a2b1aab114302946c2ac6f44fdff4e